cyfronet-fid/marketplace
# [#2139] Fix project message caption

Closes #2139. Potentially closes #1359 (to check by @michal-szostak)

- New conversations view [@jarekzet, @wujuu]
- Improved conversation labels (which now include: date, role and optionally email and name (not relevant in the context of user messages)) [@wujuu]

How to test:
1. Create a project
2. Navigate to `/projects/<project_id>/conversation`
2. Fill in and send the message form in the frontend to see the **user message**
3. Execute following lines in `rails c`:
```
> Message.create(message: "message from provider", scope: "public", author_role: "provider", author_name: "Jan Kowalski", author_email: "qwe@google.com", messageable: Project.last)
> Message.create(message: "message from customer support", scope: "public", author_role: "mediator", author_name: "Pan Kowalski", author_email: "qwe@google.com", messageable: Project.last)
> Message.create(message: "TOP SECRET message from provider", scope: "user_direct", author_role: "provider", author_name: "Pani Kowalska", author_email: "qwe@google.com", messageable: Project.last)
> Message.create(message: "message from provider", scope: "public", author_role: "provider", author_name: "Jan Kowalski", author_email: nil, messageable: Project.last)
> Message.create(message: "message from provider", scope: "public", author_role: "provider", author_name: nil, author_email: "qwe@google.com", messageable: Project.last)
> Message.create(message: "message from provider", scope: "public", author_role: "provider", author_name: nil, author_email: nil,  messageable: Project.last)
```
... to seed the conversation with:
- message from the **provider**, who is identified by **name and email**
- message from the **mediator** (customer support), who is identified by **name and email**
- message from the **provider**, who should have "VISIBLE ONLY TO YOU" label (provider identified by name and email
- message from the **provider**, who is identified by **email only**
- message from the **provider**, who is identified by **name only**
- message from the **provider**, who is **anymous** (didn't provide any additional information) 
4. _NOTE how the labels change!_
5. (optional) Mix and match `message`, `author_name`, `author_email` with different values if needed (to check how the UI behaves for longer names for example)
6. (optional) Fill in `author_role` with `provider` or `mediator` values to check for unspecified scenarios if needed (to check how the label behaves on different message lengths for example)
7. Project Item messages should look the same. To seed it in the same way, add a project item to your project and execute the above commands with the exception that the `messageable` field should be `ProjectItem.last`. Then navigate to `/projects/<project_id>/services/<project_item_id>/conversation` to see the seeded conversations.

Conversations should look like in the preview: https://invis.io/JXOZKC8WMDU#/460826937_Projects_-_Contact_Experts_-_Chat
